### PR TITLE
Storage: Cleanup object history API

### PR DESCRIPTION
**Pull request by ryantxu**
This updates the history API so it returns its own object rather than `RawObject` directly -- this also means we can move the comment/message so it lives in history exclusively rather than on the RawObject.

This merges into https://github.com/grafana/grafana/pull/56214 since that was the easiest way for me to test it :)

### CODE REVIEW

File: pkg/services/store/object/dummy/dummy_server.go [hunk 1]
```diff
@@ -158,27 +177,46 @@ func (i dummyObjectServer) update(ctx context.Context, r *object.WriteObjectRequ
...
+				ETag:       updated.ETag,
+				Comment:    r.Comment,
+			},
+		}
+		rsp.Object = versionInfo.ObjectVersionInfo
+
+		// When saving, it must be differnet than the head version
+		if i.ETag == updated.ETag {
+			versionInfo.ObjectVersionInfo.Version = i.Version
+			rsp.Error = &object.ObjectErrorInfo{
```
Comment (ArturWierzbicki) [inline pkg/services/store/object/dummy/dummy_server.go:198]: it looks more like a info-no-action-needed rather than an error 🤔

Comment (ryantxu) [reply pkg/services/store/object/dummy/dummy_server.go:198]: agree -- just not sure what better thing to show.  Perhaps just leaving out the error?  an additional status?

CREATED | UPDATED | UNCHANGED?

Comment (ArturWierzbicki) [reply pkg/services/store/object/dummy/dummy_server.go:198]: created/updated/unchanged seems good, though I think most clients just care that the operation succeeded, and the current object store data is X

Comment (ryantxu) [reply pkg/services/store/object/dummy/dummy_server.go]: I added the codes... we can remove if it feels cumbersome.

File: pkg/services/store/object/dummy/dummy_server.go [hunk 2]
```diff
@@ -74,13 +80,26 @@ func (i dummyObjectServer) findObject(ctx context.Context, uid string, kind stri
...
+				UID:        obj.UID,
+				Kind:       obj.Kind,
+				Created:    obj.Created,
+				CreatedBy:  obj.CreatedBy,
+				Modified:   objVersion.Modified,
+				ModifiedBy: objVersion.ModifiedBy,
+				ETag:       objVersion.ETag,
+				Version:    objVersion.Version,
+
+				// Not in the history object
```
Comment (ArturWierzbicki) [inline pkg/services/store/object/dummy/dummy_server.go:98]: i do not understand what this comment refers to - the `objVersion` _is_ the history object.. right? that is: https://github.com/grafana/grafana/blob/85b965cbec21637fe062572eefa8ca8f10fc102b/pkg/services/store/object/dummy/dummy_server.go#L20-L24

or is this comment referring to the gRPC `ObjectVersionInfo`?

File: pkg/services/store/object/dummy/dummy_server.go [hunk 3]
```diff
@@ -158,27 +177,44 @@ func (i dummyObjectServer) update(ctx context.Context, r *object.WriteObjectRequ
...
+				ModifiedBy: updated.ModifiedBy,
+				Size:       updated.Size,
+				ETag:       updated.ETag,
+				Comment:    r.Comment,
+			},
+		}
+		rsp.Object = versionInfo.ObjectVersionInfo
+		rsp.Status = object.WriteObjectResponse_MODIFIED
+
+		// When saving, it must be differnet than the head version
```
Comment (ArturWierzbicki) [inline pkg/services/store/object/dummy/dummy_server.go:196]:
```suggestion
		// When saving, it must be different than the head version
```

File: pkg/services/store/object/dummy/dummy_server.go [hunk 4]
```diff
@@ -263,15 +315,15 @@ func (i dummyObjectServer) History(ctx context.Context, r *object.ObjectHistoryR
 		return nil, err
 	}
 
-	if obj == nil {
-		return &object.ObjectHistoryResponse{
-			Object: nil,
-		}, nil
+	rsp := &object.ObjectHistoryResponse{}
+	if obj != nil {
+		// Return the most recent versions first
```
Comment (ArturWierzbicki) [inline pkg/services/store/object/dummy/dummy_server.go:320]: 👍 👍

**Review by ArturWierzbicki — APPROVED** [review]
🚄 🚀

**Pull request merged**